diff --git a/third_party/blink/renderer/core/probe/core_probes.pidl b/third_party/blink/renderer/core/probe/core_probes.pidl
index a91ba14dbb4f..7cf1b57dea55 100644
--- a/third_party/blink/renderer/core/probe/core_probes.pidl
+++ b/third_party/blink/renderer/core/probe/core_probes.pidl
@@ -169,7 +169,7 @@ interface CoreProbes {
 
   CallFunction([Keep] ExecutionContext* context, const v8::Local<v8::Context>& v8_context, v8::Local<v8::Function> function, int depth = 0);
   UserCallback([Keep] ExecutionContext* context, const char* name, const AtomicString& atomic_name, bool recurring, EventTarget* event_target = nullptr, Event* event = nullptr, EventListener* listener = nullptr);
-  InvokeCallback([Keep] ExecutionContext* context, const char* name, CallbackFunctionBase* callback, const v8::Local<v8::Function>& function = v8::Local<v8::Function>());
+  InvokeCallback([Keep] ExecutionContext* context, const char* name, CallbackFunctionBase* callback, const v8::Local<v8::Function> function = v8::Local<v8::Function>());
   InvokeEventHandler([Keep] ExecutionContext* context, EventTarget* event_target = nullptr, Event* event = nullptr, EventListener* listener = nullptr);
   V8Compile([Keep] ExecutionContext* context, const String& file_name, int line, int column);
   ParseHTML(Document* document, HTMLDocumentParser* parser);
